src/services/game-logic/pieces/Pawn.ts

```diff
@@ -29,21 +23,35 @@ class Pawn extends Piece {
         if (move) moves.push(move);
         if (specialMove) moves.push(specialMove);
 
+        this.setCapturingSquares(moves);
+
         return moves;
     }
 
-    /**
-     * Prepares target Square or returns null in case if move would be outside board.
-     */
-    private prepareMove(step: number): Square | null {
-        const move = new Square(this.position.row + step, this.position.column);
+    setCapturingSquares(moves: Square[]): void {
+        // Assume that white plays on bottom
+        // row = 0, column = 0 means left upper corner
+        const moveDirection = this.color === Colors.WHITE ? -1 : 1;
+        const capture1 = this.prepareMove(moveDirection, -1);
+        const capture2 = this.prepareMove(moveDirection, 1);
+
+        if (capture1) moves.push(capture1);
+        if (capture2) moves.push(capture2);
+    }
+
+    private prepareMove(rowStep: number, colStep: number = 0): Square | null {
```

Comment:
colStep jest używane tylko w setCapturingSquares także nie ma potrzeby nadawania domyślnej wartości 0 bo nigdy to nie wystąpi?

Reply:
w getPossibleMoves też używam tej metody, ale już bez ustalenia colStep, dlatego jest ustawiona wartość domyślna - zero